src/Raven.Client/Documents/Session/EntityToBlittable.cs

```diff
@@ -452,12 +452,34 @@ private static bool ShouldSimplifyJsonBasedOnType(string typeValue)
             return typeof(IEnumerable).IsAssignableFrom(type.GetGenericTypeDefinition());
         }
 
-        private static Type GetPropertyType(string propertyName, Type rootType)
+        internal static Type GetPropertyType(string propertyName, Type rootType)
         {
             if (rootType == null)
                 return null;
 
-            var memberInfo = ReflectionUtil.GetPropertyOrFieldFor(rootType, BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic, propertyName);
+            MemberInfo memberInfo = null;
+            try
+            {
+                memberInfo = ReflectionUtil.GetPropertyOrFieldFor(rootType, BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic, propertyName);
+            }
+            catch (AmbiguousMatchException)
+            {
+                var memberInfos = ReflectionUtil.GetPropertiesAndFieldsFor(rootType, BindingFlags.Public | BindingFlags.Instance | BindingFlags.NonPublic)
+                    .Where(x => x.Name == propertyName)
+                    .ToList();
+
+                while (typeof(object) != rootType)
+                {
+                    memberInfo = memberInfos.FirstOrDefault(x => x.DeclaringType == rootType);
```

Comment:
IsAssignableFrom, instead?

Reply:
we want to get 'closest' decendant and IsAssignableFrom returns any, or I'm wrong?

Reply:
But you are missing interfaces

Reply:
Added test with classes that inherit from interface